src/utils/resolveWithRTL.js

```diff
@@ -19,24 +18,18 @@ export default function resolveWithRTL(css, styles) {
     inlineStyles,
   } = separateStyles(flattenedStyles);
 
-  const stylesWithDirection = aphroditeStyles.map((stylesObj) => {
-    let definition = stylesObj._definition;
+  aphroditeStyles.forEach((stylesObj) => {
+    // The _definition key is an implementation detail of aphrodite. If aphrodite
+    // changes it in the future, this code will need to be updated.
+    const definition = stylesObj._definition;
     const directionalStyles = generateDirectionalStyles(definition);
-    if (directionalStyles) {
-      const { sharedStyles, ltrStyles, rtlStyles } = directionalStyles;
-      definition = {
-        ...sharedStyles,
-        [LTR_SELECTOR]: ltrStyles,
-        [RTL_SELECTOR]: rtlStyles,
-      };
-    }
 
-    return {
-      ...stylesObj,
-      _definition: definition,
-    };
+    if (!directionalStyles) return;
+
+    stylesObj._definition = directionalStyles; // eslint-disable-line no-param-reassign
   });
 
+  const stylesWithDirection = [...aphroditeStyles];
```

Comment:
I don't think I understand why you need to copy aphroditeStyles here. Can you explain?